### PR TITLE
Dockerfile for rest-services

### PR DESCRIPTION
**Pull request by Jayalakshmi02**
Added production.js and development.js
Created Dockerfile

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,5 @@
+POSTGRE_HOST = "localhost"
```
Comment (prepaka) [inline .env:1]: Why this file is needed?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
+FROM node:12.16.2-alpine3.10
+
+RUN apk add --virtual builds-deps build-base
+RUN apk add python make gcc
```
Comment (prepaka) [inline Dockerfile:4]: Why we need python make gcc packages?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,25 @@
+FROM node:12.16.2-alpine3.10
+
+RUN apk add --virtual builds-deps build-base
```
Comment (prepaka) [inline Dockerfile:3]: can you add these --update --no-cache.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ENV POSTGRE_HOST = "localhost"\
+    POSTGRE_DB = "inventory"\
+    POSTGRE_PORT = "5432"\
+    POSTGRE_USERNAME = "vento"\
+    POSTGRE_PASSWORD = "control1234"
+
+COPY yarn.lock yarn.lock
+COPY package.json package.json
+
+RUN yarn add sails --network-timeout 10000000
```
Comment (prepaka) [inline Dockerfile:15]: Any specific reason for network-timeout?

File: config/datastores.js [hunk 5]
```diff
@@ -12,7 +12,7 @@
  * For more information on configuring datastores, check out:
  * https://sailsjs.com/config/datastores
  */
-
+const c = require('./env/key.js')
```
Comment (prepaka) [inline config/datastores.js:15]: As suggested over call please change it.

File: config/env/key.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+if (process.env.NODE_ENV === "production") {
```
Comment (prepaka) [inline config/env/key.js:1]: As suggested over call please remove this file.

File: config/env/production.js [hunk 7]
```diff
@@ -1,397 +1,18 @@
...
+  PostgreDb : {
+    host: process.env.POSTGRE_HOST || 'localhost',
+    port: process.env.POSTGRE_PORT || 5432,
+    db: process.env.POSTGRE_DB || 'inventory',
+    username: process.env.POSTGRE_USERNAME || 'vento',
+    password: process.env.POSTGRE_PASSWORD || 'control1234'
+  }
+},
+sockets: {
+  onlyAllowOrigins: [
```
Comment (prepaka) [inline config/env/production.js:13]: Is it needed?

File: Dockerfile (hunk 4 above)
Comment (DhyaniKanika) [reply Dockerfile:15]: Sir our networks are very slow so we had to increase the default network timeout. Without it, it was throwing errors.

File: Dockerfile (hunk 2 above)
Comment (DhyaniKanika) [reply Dockerfile:4]: Sir the alpine node is very small so it doesn't have python and gcc in it. For npm install and installing other packages too we need python so we had to add them.

Comment (DhyaniKanika) [reply Dockerfile:4]: Sir the alpine node is very small so it doesn't have python and gcc in it. For npm install and installing other packages too we need python so we had to add them.

File: config/env/production.js (hunk 7 above)
Comment (Jayalakshmi02) [reply config/env/production.js]: Screenshot (154)
sir we are getting the above error if it is not set

**Pull request merged**